src/Draco.FormatterEngine/TokenMetadata.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Draco.Compiler.Internal.Syntax.Formatting;
+
+/// <summary>
+/// Represent the formatting metadata of a token.
+/// </summary>
+/// <param name="Kind">The whitespace behavior of this token.</param>
+/// <param name="Text">The text of the token.</param>
+/// <param name="DoesReturnLine">
+/// <para>
+///     An optional reference to a nullable boolean, indicate whether the line returns or not.
+/// </para>
+/// <para>
+///     If the field is null, it means there was no decision was made on this token, we default to not returning a line.
+///</para>
+/// <para>
+///     If the field is set to a reference, but the value of the Box null, it's because this is a pending
+/// </para>
```

Comment:
This is a bit of a primitive obsession, isn't there a nicer way to represent this?
For example, should the box include the nullability of its value, just because it's not computed yet? (`Task` comes to my mind)